src/Microsoft.AspNet.Mvc.Core/Builder/RouteBuilderExtensions.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Routing;
+
+namespace Microsoft.AspNet.Builder
+{
+    /// <summary>
+    /// Extension methods for <see cref="IRouteBuilder"/>.
+    /// </summary>
+    public static class RouteBuilderExtensions
+    {
+        /// <summary>
+        /// Adds a route to the <see cref="IRouteBuilder"/> with the given MVC area with the specified
+        /// name and template.
```

Comment:
> Separately did you intend to create an overload that only had areaName and template parameters i.e. used a default route name?

No, `MapRoute` always has a positional parameter for route name, even though it's not required.